ci/authn-gce/get_gce_id_tokens.sh

```diff
@@ -0,0 +1,151 @@
+    # - instance status: used to validate that the instance is running.
+    #	----------------------------------------------------------------
+    set_instance_zone "$instance_info"
+    local instance_status=$(echo "$instance_info" | cut -f2 -d ';')
+
+    if [ "$instance_status" = "RUNNING" ]; then
+      echo "-- GCE instance '${instance_name}', zone: '$INSTANCE_ZONE', status: '$instance_status'."
+      INSTANCE_RUNNING=1
+    else
+      echo "-- GCE instance '${instance_name}', zone: '$INSTANCE_ZONE', NOT RUNNING!, status: '$instance_status'."
+      print_running_gce_instances
+      error_exit "Instance is not running."
+    fi
+  else
+    echo "-- GCE instance '${instance_name}' NOT FOUND!"
+    print_running_gce_instances
+    error_exit "Instance not found."
+  fi
+}
+
+set_instance_zone() {
+  local instance_info="$1"
+  INSTANCE_ZONE=$(echo "$instance_info" | cut -f1 -d ';')
+}
+
+print_running_gce_instances() {
+  echo "-- Retrieving running instances..."
+  gcloud compute instances list --limit=50\
+    --format="table[box,title='Running Compute Instances'](name,machine_type.basename(),status)" --filter="STATUS=RUNNING"
+}
+
+get_tokens_into_files() {
+  if [ "${INSTANCE_EXISTS}" = "0" ] | [ "${INSTANCE_RUNNING}" = "0" ]; then
+    error_exit "-- Cannot run command, GCE instance '${INSTANCE_NAME}' not in a valid state!"
+  fi
+
+  get_token_into_file "full" "conjur/cucumber/host/test-app" "gce_token_valid"
+  get_token_into_file "full" "conjur/cucumber/host/non-existing" "gce_token_non_existing_host"
+  get_token_into_file "full" "conjur/cucumber/host/non-rooted/test-app" "gce_token_non_rooted_host"
+  get_token_into_file "full" "conjur/cucumber/test-app" "gce_token_user"
+  get_token_into_file "full" "conjur/non-existing/host/test-app" "gce_token_non_existing_account"
+  get_token_into_file "full" "invalid_audience" "gce_token_invalid_audience"
+  get_token_into_file "standard" "conjur/cucumber/host/test-app" "gce_token_standard_format"
+  wait
+}
+
+get_token_into_file() {
+  local token_format="$1"
+  local audience="$2"
+  local filename="$3"
+  local curl_cmd="curl -s -G -H 'Metadata-Flavor: Google' \
+  --data-urlencode 'format=${token_format}' \
+  --data-urlencode 'audience=${audience}' \
+  'http://metadata/computeMetadata/v1/instance/service-accounts/default/identity'"
+  echo "-- Obtain an ID token and write the token in ${token_format} format with audience: ${audience} to \
+  ${TOKENS_OUT_DIR_PATH}/${filename}"
+  gcloud compute ssh "$INSTANCE_NAME" --zone="${INSTANCE_ZONE}" --command "${curl_cmd}" > "${TOKENS_OUT_DIR_PATH}/${filename}" &
+}
+
+error_exit() {
+  #	----------------------------------------------------------------
+  #	Function for exit due to fatal program error
+  #		Accepts 1 argument:
+  #			string containing descriptive error message
+  #	----------------------------------------------------------------
+
+  echo "${PROGNAME}: ${1:-"Unknown Error"}" 1>&2
+  exit 1
+  echo "$1" 1>&2
```

Comment:
why are we suppressing the error with `&2` and returning our own if we anyways exit with 1?

Reply:
I think it is just redundant, good catch!